ports/geckolib/glue.rs

```diff
@@ -3859,7 +3859,8 @@ pub extern "C" fn Servo_HasPendingRestyleAncestor(element: RawGeckoElementBorrow
     let mut element = Some(GeckoElement(element));
     while let Some(e) = element {
         if let Some(data) = e.borrow_data() {
-            if data.restyle.hint.has_non_animation_invalidations() {
+            if data.restyle.hint.has_non_animation_invalidations() ||
+               data.restyle.hint.has_animation_hints() {
```

Comment:
Per http://logs.glob.uno/?c=mozilla%23servo&s=12+Sep+2017&e=12+Sep+2017#c750671 I think this should check instead `!data.restyle.hint.is_empty()`. instead.